### PR TITLE
feat(agent): curator multimethod + :merge-resolution (Stage 2A)

### PR DESCRIPTION
**Pull request by H31MDALLR**
## Summary

Stage 2A of v2 multi-parent merge per [I-DAG-MULTI-PARENT-MERGE.md §6.1.2](https://github.com/miniforge-ai/miniforge/blob/main/specs/informative/I-DAG-MULTI-PARENT-MERGE.md). Refactors `agent/curator` from a single function into a multimethod and adds the `:merge-resolution` method that the resolution sub-workflow (Stage 2B) will call between iterations.

After this PR, the curator surface supports the merge-resolution check pipeline; **Stage 2B will land the sub-workflow itself and wire it to `merge-parent-branches!`'s conflict path.**

## What's in this PR

**`curate` multimethod** dispatched on `:curator/kind`:

| Kind | Behavior |
| ---- | -------- |
| `:implement` (default) | Existing logic — produces a CuratedArtifact, terminal `:curator/no-files-written` on empty diff |
| `:merge-resolution` (new) | Validates an iteration of the resolution sub-workflow per spec §6.1.2 |

**Merge-resolution method** runs two pure-data checks against the worktree:

- **`:curator/markers-not-resolved`** — git's conflict markers (`<<<<<<<`, `=======`, `>>>>>>>`) are still in the worktree. The agent terminated but didn't finish the job.
- **`:curator/recurring-conflict`** — the conflicted path set this iteration matches the prior iteration's. Agent isn't making progress; the resolution loop should terminate before exhausting the rest of the budget. Per spec §6.1.2 this is the "agent stuck" early-out.

Both checks operate on file-scan + path-set comparison; no agent narration needed.

**Backward compatibility**: `curate-implement-output` is now a thin wrapper around `(curate (assoc input :curator/kind :implement))`. Existing callers (`phase-software-factory.implement`) keep working unchanged. The interface re-exports both `curate` (new entry point) and `curate-implement-output` (legacy alias).

## Test plan

- [x] 6 new tests / 13 assertions in `curator_merge_resolution_test.clj` covering:
  - Clean worktree → success (`:resolution/markers-cleared?` true)
  - Markers present (first iteration) → `:curator/markers-not-resolved`
  - Same path set as prior → `:curator/recurring-conflict`
  - Different path set as prior → `:curator/markers-not-resolved` (not recurring; agent IS making progress)
  - Nil prior-paths → first-iteration behavior
  - Default dispatch (no `:curator/kind`) → `:implement` legacy contract preserved
- [x] Full `bb test` suite: 5011 / 22902 / 0 / 0
- [ ] Stage 2B will exercise the curator method end-to-end against a real conflicted git repo with the resolution sub-workflow loop

## Spec traceability

| Spec section | Implemented in this PR |
| ------------ | ---------------------- |
| §6.1.2 `:curator/markers-not-resolved` check | `markers-not-resolved-error` factory + scan |
| §6.1.2 `:curator/recurring-conflict` check | `recurring-conflict-error` factory + path-set comparison |
| §10.12 curator extension via existing polymorphism | Multimethod refactor; `:implement` keeps legacy contract |

## What's NOT in this PR (deferred to Stage 2B)

- The resolution sub-workflow definition that calls `(curate (assoc input :curator/kind :merge-resolution))` between iterations
- Wiring `merge-parent-branches!`'s conflict path to spawn the resolution sub-workflow
- Policy override hook for the resolution agent prompt (spec §6.1.3)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: components/agent/src/ai/miniforge/agent/curator.clj [hunk 1]
```diff
@@ -395,6 +396,139 @@
...
+      (and (set? prior-conflicted-paths)
+           (= prior-conflicted-paths current-paths))
+      (recurring-conflict-error worktree-path current-paths)
+
+      ;; Markers still present but the path set has changed (some
+      ;; resolved, others surfaced, or the conflict shape shifted).
+      ;; The loop should re-prompt; this is a "keep going" terminal
+      ;; for THIS iteration, not the whole loop.
+      :else
+      (markers-not-resolved-error worktree-path current-paths))))
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/agent/src/ai/miniforge/agent/curator.clj:583]: Fixed in 51633125 — scan-conflicted-paths still returns nil for invalid input, but the :merge-resolution method now branches on nil first and surfaces a typed :curator/worktree-missing fault rather than letting (empty? nil) → true silently report success. New :error/curator-worktree-missing catalog entry; new test merge-resolution-worktree-missing-test pins both the non-existent-path and nil-path cases.

[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/agent/src/ai/miniforge/agent/curator.clj:583]: Fixed in 51633125 — prior-conflicted-paths is now coerced to a set with (set prior-conflicted-paths) when non-nil. The curator's own :conflicted-paths sorted-vector output round-trips correctly into the next iteration's recurrence detection. nil still preserves first-iteration semantics. New test merge-resolution-prior-paths-coerced-from-vector-test exercises feeding back a vector.

File: components/agent/src/ai/miniforge/agent/curator.clj [hunk 2]
```diff
@@ -395,6 +396,139 @@
...
+;;
+;; Both checks operate on pure-data inspections of the worktree state
+;; (file scan + path-set comparison). They don't need the agent's
+;; narration to fire.
+
+(def ^:private conflict-marker-re
+  "Match a line that begins (after optional whitespace) with one of
+   git's three conflict-marker tokens. The exact-7 form is git's
+   default; longer runs are also legal so we match `+`."
+  #"(?m)^[<>=]{7,}")
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/agent/src/ai/miniforge/agent/curator.clj]: Fixed in 51633125 — regex now ^(?:<{7,}|={7,}|>{7,}) (three alternatives, each a single-character run). The mixed-character class would have false-positived on lines like '<>=<>=<' which aren't conflict markers. Docstring updated to describe the actual behavior accurately (column 0, single-character run, 7+ length to match git's emit format). New test conflict-marker-regex-only-matches-real-tokens-test pins that mixed and short runs are NOT flagged.

File: components/agent/src/ai/miniforge/agent/curator.clj [hunk 3]
```diff
@@ -395,6 +396,139 @@
...
+  "Walk `worktree-path` and return the set of relative paths whose file
+   contents contain conflict markers. The return is a set so callers
+   can compare iterations directly."
+  [worktree-path]
+  (when (fs/directory? worktree-path)
+    (let [root (fs/canonicalize worktree-path)]
+      (->> (fs/glob root "**" {:hidden false})
+           (filter file-has-conflict-markers?)
+           (map (fn rel-path [p] (str (fs/relativize root p))))
+           set))))
```
[the remark this answers is not in the document]
Comment (H31MDALLR) [reply components/agent/src/ai/miniforge/agent/curator.clj]: Fixed in 51633125 — two improvements stacked. Primary path: try git grep first via shell/sh, scoped to tracked files only. Skips ignored / vendor / build dirs naturally. v2 merge-resolution worktrees are always git-managed (created by merge-parent-branches! via git worktree add) so this is the common path. Fallback path (non-git tree, git missing): walks the worktree but with line-streamed reads via clojure.java.io/reader + line-seq + some + early-exit. Files no longer load fully into memory just to scan for one line.
